extensions/guess.py

```diff
@@ -0,0 +1,61 @@
+import discord
+from discord.ext import commands
+import random
+
+class Guess(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        # The number the user is trying to guess
+        self.number = self.generate_number()
+        # Number of guesses made at getting the right number
+        self.attempts = 0
+        # Track who is actually playing the game.
+        self.guesser = ''
+        # Track if the game is active.
+        self.active = False
+
+    def generate_number(self):
+        return random.randint(0, 100)
+
+    @commands.command(name='guess')
+    async def guess(self, ctx):
+        """Guess the number"""
+        self.guesser = ctx.author
+        self.active = True
+        await ctx.send('Starting guessing game with {}'.format(ctx.author))
+
+    @commands.Cog.listener()
+    async def on_message(self, ctx):
+        if ctx.author == self.guesser and self.active == True and ctx.content.split(' ')[0] != '.guess':
```

Comment:
The `guess` command name should be moved to a constant since it's referenced in multiple places. The command prefix(es) you should be able to extract from the `bot` or the `ctx` APIs. If not, then perhaps we should create a dictionary directly on the `bot` variable in main.py and store the command prefixes there.